engine/runtime-compiler/src/main/scala/org/enso/compiler/data/BindingsMap.scala

```diff
@@ -77,23 +105,55 @@ case class BindingsMap(
     *                  instances
     * @return `this` with module references converted to concrete
     */
-  def toConcrete(moduleMap: ModuleMap): Option[BindingsMap] = {
-    val newMap = this.currentModule.toConcrete(moduleMap).map { module =>
-      this.copy(currentModule = module)
+  private def ensureConvertedToConcrete(): Option[BindingsMap] = {
+    val r = pendingRepository
+    if (r != null) {
+      val res = toConcrete(r, r.getModuleMap).map { b =>
+        pendingRepository     = null
+        this._currentModule   = b._currentModule
+        this._exportedSymbols = b._exportedSymbols
+        this._resolvedImports = b._resolvedImports
+        this
+      }
+      res
```

Comment:
```suggestion
      toConcrete(r, r.getModuleMap).map { b =>
        pendingRepository     = null
        this._currentModule   = b._currentModule
        this._exportedSymbols = b._exportedSymbols
        this._resolvedImports = b._resolvedImports
        this
      }
```

_spurious_

Reply:
Often I'd like to investigate the computed value on return - with the additional `res` variable such debugging is easier.

Reply:
Changed in 63e875d3146638cf8aabc8216c115992f7567a2c